### PR TITLE
Move constants and types that could shared in the package v1alpha1

### PR DESCRIPTION
**Pull request by ArielLima**
Ticket: https://issues.redhat.com/browse/OSD-2835

Create pkg/apis/aws/v1alpha1/shared_types.go add constants and types that could shared in the package v1alpha1

https://github.com/openshift/aws-account-operator/blob/master/pkg/controller/account/account_controller.go#L79-L138

Done criteria:
- Move types and constants that are shared between APIs to this file
- Any constants that we can reference here to keep controller files clean

Testing:
- To test you just need to apply  an account CR and check that it becomes ready without issues.

Edit: Description

### CODE REVIEW

**Comment by ArielLima** [comment]
Is it reasonable to move errors out as well? Are we doping this to every controller with this PR?
cc @jharrington22 @rogbas

**Comment by ArielLima** [comment]
**Tested with new make tests**

File: pkg/apis/shared_types.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package apis
```
Comment (fahlmant) [inline pkg/apis/shared_types.go]: Is there a reason this isn't put into the utils package or something similar? This is not a very descriptive name IMO

Comment (ArielLima) [reply pkg/apis/shared_types.go]: Simply followed the ticket, I can move it though. What do you think? @rogbas @jharrington22

Comment (fahlmant) [reply pkg/apis/shared_types.go]: The tickets specifies the pkg `v1alpha1` as I read it.

Comment (ArielLima) [reply pkg/apis/shared_types.go]: https://issues.redhat.com/browse/OSD-2835 @fahlmant

Comment (jharrington22) [reply pkg/apis/shared_types.go]: Yes apis isn't quite right. I wanted it to be part of the v1alpha1 package I just got the path wrong in the ticket. @ArielLima please update.

File: pkg/controller/account/account_controller.go [hunk 2]
```diff
@@ -29,6 +28,7 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 
+	"github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
 	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
```
Comment (iamkirkbater) [inline pkg/controller/account/account_controller.go]: Is there a reason we're importing this again?

File: pkg/controller/account/iam.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/aws/aws-sdk-go/service/sts"
 	"github.com/go-logr/logr"
+	"github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
 	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
```
Comment (iamkirkbater) [inline pkg/controller/account/iam.go]: Same question, we're importing the same package twice here.

File: pkg/controller/account/account_controller.go (hunk 2 above)
Comment (ArielLima) [reply pkg/controller/account/account_controller.go]: No, fixed

File: pkg/controller/account/iam.go (hunk 3 above)
Comment (ArielLima) [reply pkg/controller/account/iam.go]: No, fixed

**Comment by fahlmant** [comment]
@ArielLima Can you update the PR title and the commit message?

**Comment by ArielLima** [comment]
> @ArielLima Can you update the PR title and the commit message?

:+1:

**Pull request merged**